src/OrbitBase/ProcessIdLinux.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <filesystem>
+#include <optional>
+#include <system_error>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ProcessId.h"
+#include "OrbitBase/Result.h"
+#include "absl/strings/numbers.h"
+
+namespace orbit_base {
+
+namespace fs = std::filesystem;
+
+namespace {
+
+std::optional<pid_t> ProcEntryToPid(const std::filesystem::directory_entry& entry) {
+  if (!entry.is_directory()) {
+    return std::nullopt;
+  }
+
+  int potential_pid;
+  if (!absl::SimpleAtoi(entry.path().filename().string(), &potential_pid)) {
+    return std::nullopt;
+  }
+
+  if (potential_pid <= 0) {
+    return std::nullopt;
+  }
+
+  return static_cast<pid_t>(potential_pid);
```

Comment:
is this static cast needed here?

Reply:
Not in practice, but `potential_pid` is `int`, not `pid_t`.

Reply:
I guess it can be made `pid_t`? I'm not sure why it was like that.

Reply:
I think it is safer to remove this. we would want to know if pid_t becomes something that needs an explicit cast from int

Reply:
done.